### ISSUE TITLE
interpolated resize has the wrong size

### ISSUE THREAD

**Issue by joseluis** [dankamongmen/notcurses#1797]
using the pixel-cell.rs example:, it shows the interpolated resize is either smaller (alacritty, wezterm) or larger (xterm) than it should be.

in latest alacritty:
![image](https://user-images.githubusercontent.com/5832068/122641147-ae0b1100-d103-11eb-8225-8a03968da579.png)

in xterm:
![image](https://user-images.githubusercontent.com/5832068/122641222-2a9def80-d104-11eb-9975-19a0eb5be5c5.png)

in latest nightly wezterm:
![image](https://user-images.githubusercontent.com/5832068/122641252-65a02300-d104-11eb-93a5-76f1f5147012.png)

can't test in kitty because #1796

**Comment by dankamongmen** [comment]
i assume this is recent breakage? if so, it's probably related to #1789's fix. need better unit testing!

**Comment by dankamongmen** [comment]
i confirm your behavior. i have a unit test on this stuff, i think, so it's odd that we're seeing this. let me try and trace it down.

**Comment by dankamongmen** [comment]
check out the PoC i've added, `interp`. it does something very similar. is it working (i.e. all "squares" are the same size) for you? i just put a bunch of comments into #1796 that really belonged here, argh.

**Comment by dankamongmen** [comment]
looking at `pixel-cell.rs`, i see:

```rs
    // resize the ncvisual (doesn't use interpolation)                                                     
    let voptions4 =                                                                                        
        NcVisualOptions::without_plane(7, 37, 0, 0, pg.cell_y, pg.cell_x, NCBLIT_PIXEL, 0, 0);             
    v1.resize_noninterpolative(pg.cell_y * 4, pg.cell_x * 4)?;                                             
    v1.render(&mut nc, &voptions4)?;                                                                       
    rsleep![&mut nc, 1];                                                                                   
                                                                                                           
    // resize the ncvisual (uses interpolation)                                                            
    let v5 = NcVisual::from_rgba(buffer.as_slice(), pg.cell_y, pg.cell_x * 4, pg.cell_x)?;                 
    let voptions5 =                                                                                        
        NcVisualOptions::without_plane(7, 51, 0, 0, pg.cell_y, pg.cell_x, NCBLIT_PIXEL, 0, 0);             
    v5.resize(18 * 4, 9 * 4)?; // FIXME: render function fails when downsizing (y<18 | x<9) 
```

in one you've got `pg.cell_{y,x} * 4` and the other you've got `{18,9} * 4` as your resize targets. why would they be the same size?

**Comment by dankamongmen** [comment]
> in one you've got `pg.cell_{y,x} * 4` and the other you've got `{18,9} * 4` as your resize targets. why would they be the same size?

and we would expect them to change with the cell-pixel geometry, i.e. with different terminals, which is exactly what we see...no?

**Comment by joseluis** [comment]
ok I've confirmed your poc works in all terminals, I'll fix my example, sorry about that

**Comment by dankamongmen** [comment]
> ok I've confirmed your poc works in all terminals, I'll fix my example, sorry about that

well, it might not necessarily be your sample -- it might be the particular geometries you're using, etc. let's leave this open until we know for sure (unless you agree with my assessment above). feel free to play around with or look at that `interp.c` in `src/poc`.

**Comment by joseluis** [comment]
Don't worry! It was what you said, I forgot those contant values in there... it's fixed in 90a2ff6fcaad4df653e0b4957b0846b4d47d8991
